x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/DocVector.java

```diff
@@ -173,7 +173,21 @@ public DocBlock asBlock() {
 
     @Override
     public DocVector filter(int... positions) {
-        return new DocVector(shards.filter(positions), segments.filter(positions), docs.filter(positions), null);
+        IntVector filteredShards = null;
```

Comment:
Great catch!

Reply:
It took so long to find this one! `git bisect` and then running the test a bunch and just staring at exception. Took all day...

Reply:
Ah, for me, to find a leak, I copied the MockBlockFactory to the esql module and ran tests with it. It will show the leak.

Reply:
Oh. That's a good trick!

There's another leak detector in `server` that folks have started to use for netty things. I want to pull it in on our side too at some point.